File: scm-java/src/main/scheme/primitives/PrimitiveExceptionHandlersSet.java
package scheme.primitives;
import scheme.*;

public class PrimitiveExceptionHandlersSet extends Primitive {
    @Override public String name() { return "%exception-handlers-set!"; }
    @Override public String info() {
        return "Syntax: (%exception-handlers-set! handlers)\n" +
               "Library: (scm core)\n" +
               "Description: Internal primitive. Sets the current VM's exception handler stack to handlers.\n" +
               "Example:\n" +
               "  (%exception-handlers-set! '())";
    }
    @Override public Object apply(SourcePos pos, Object[] args) {
        checkArgs(pos, args, 1, 1);
        VM vm = VM.current.get();
        if (vm != null) vm.exceptionHandlers = args[0];
        return Value.NIL;
    }
}

File: scm-csharp/scheme/primitives/PrimitiveExceptionHandlersSet.cs
namespace scheme;

class PrimitiveExceptionHandlersSet : Primitive {
    public override string Name() => "%exception-handlers-set!";
    public override string Info() =>
        "Syntax: (%exception-handlers-set! handlers)\n" +
        "Library: (scm core)\n" +
        "Description: Internal primitive. Sets the current VM's exception handler stack to handlers.\n" +
        "Example:\n" +
        "  (%exception-handlers-set! '())";
    public override object Apply(SourcePos? pos, object[] args) {
        CheckArgs(pos, args, 1, 1);
        if (VM.Current != null) VM.Current.exceptionHandlers = args[0];
        return Value.NIL;
    }
}
